### PR TITLE
Schemas Part 1: parsing support

### PR DESCRIPTION
**Pull request by dburkart**
Progress towards #17

Things still to do (for this PR):

- [x] Full Grammar support

Things still to do for issue #17 after this PR:

- Ability to associate a schema with a topic tree
- Enforce a schema on incoming data

### CODE REVIEW

File: pkg/schema/scanner.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+			skip = s.SkipToBoundary(isDelimiter)
+		case r == 's':
+			if strings.HasPrefix(s.Input[s.Pos:], "string") {
+				t.Type = TOK_TYPE
+				skip = len("string")
+				break
+			}
+			t.Type = TOK_INVALID
+			skip = s.SkipToBoundary(isDelimiter)
+		case r == 'u':
```
Comment (gideonw) [inline pkg/schema/scanner.go:159]: Is there supposed to be an f for float here?

File: pkg/schema/scanner_test.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	n = s.MatchKey()
+	if n != 0 {
+		t.Fail()
+	}
+}
+
+func TestScannerTypes(t *testing.T) {
+	s := Scanner{Input: "boolean int8 int16 int32 int64 string"}
+
+	expectedKeywordLexemes := []string{"boolean", "int8", "int16", "int32", "int64", "string"}
```
Comment (gideonw) [inline pkg/schema/scanner_test.go:53]: Same here for floats

**Review by gideonw — COMMENTED** [review]
Float seems to be missing from the scanner?

File: pkg/schema/scanner.go (hunk 1 above)
Comment (dburkart) [reply pkg/schema/scanner.go:159]: +1 good catches

**Comment by dburkart** [comment]
Addressed review comments, merging.

**Pull request merged**